src/server/tiering/disk_storage.cc

```diff
@@ -77,6 +76,10 @@ std::error_code DiskStorage::Open(std::string_view path) {
 }
 
 void DiskStorage::Close() {
+  using namespace std::chrono_literals;
+  while (pending_ops_ > 0)
```

Comment:
nit: it's a potential deadlock point. I suggest to use `LOG_EVERY_T` to at least bring visibility in case it gets stuck there.

Reply:
Added to my to-do list, DEBUG STACKTRACE should point it out easily if it does

Reply:
debug stacktrace won't work during the shutdown flow